x/staking/keeper/liquid_stake.go

```diff
@@ -0,0 +1,155 @@
+// Returns true if the cap is exceeded
+func (k Keeper) CheckExceedsValidatorBondCap(ctx sdk.Context, validator types.Validator, shares sdk.Dec) bool {
+	validatorBondFactor := k.ValidatorBondFactor(ctx)
+	maxValLiquidShares := validator.TotalValidatorBondShares.Mul(validatorBondFactor)
+	return validator.TotalLiquidShares.Add(shares).GT(maxValLiquidShares)
+}
+
+// SafelyIncreaseTotalLiquidStakedTokens increments the total liquid staked tokens
+// if the global cap is enabled and is not surpassed by this delegation
+func (k Keeper) SafelyIncreaseTotalLiquidStakedTokens(ctx sdk.Context, amount sdk.Int) error {
+	// If the cap is disabled, do nothing
+	if !k.GlobalLiquidStakingCapEnabled(ctx) {
+		return nil
+	}
+
+	// Confirm the cap will not be exceeded
+	if k.CheckExceedsGlobalLiquidStakingCap(ctx, amount) {
+		return types.ErrGlobalLiquidStakingCapExceeded
+	}
+
+	// Increment the global total liquid staked
+	k.SetTotalLiquidStakedTokens(ctx, k.GetTotalLiquidStakedTokens(ctx).Add(amount))
+
+	return nil
+}
+
+// DecreaseTotalLiquidStakedTokens decrements the total liquid staked tokens
+// if the global cap is enabled
+func (k Keeper) DecreaseTotalLiquidStakedTokens(ctx sdk.Context, amount sdk.Int) {
+	if k.GlobalLiquidStakingCapEnabled(ctx) {
+		k.SetTotalLiquidStakedTokens(ctx, k.GetTotalLiquidStakedTokens(ctx).Sub(amount))
+	}
+}
+
+// SafelyIncreaseValidatorTotalLiquidShares increments the total liquid shares on a validator
+// if the validator bond factor is enabled and is not surpassed by this delegation
+func (k Keeper) SafelyIncreaseValidatorTotalLiquidShares(ctx sdk.Context, validator types.Validator, shares sdk.Dec) error {
+	// If the cap is disabled, do nothing
+	if !k.ValidatorBondFactorEnabled(ctx) {
+		return nil
+	}
+
+	// Confirm the validator bond factor will be not exceeded
+	if k.CheckExceedsValidatorBondCap(ctx, validator, shares) {
+		return types.ErrInsufficientValidatorBondShares
+	}
+
+	// Increment the validator's total liquid shares
+	validator.TotalLiquidShares = validator.TotalLiquidShares.Add(shares)
+	k.SetValidator(ctx, validator)
+
+	return nil
+}
+
+// DecreaseValidatorTotalLiquidShares decrements the total liquid shares on a validator
+// if the validator bond factor is enabled
+func (k Keeper) DecreaseValidatorTotalLiquidShares(ctx sdk.Context, validator types.Validator, shares sdk.Dec) {
+	if k.ValidatorBondFactorEnabled(ctx) {
+		validator.TotalLiquidShares = validator.TotalLiquidShares.Sub(shares)
+		k.SetValidator(ctx, validator)
+	}
+}
+
+// SafelyDecreaseValidatorBond decrements the total validator's self bond
+// so long as it will not cause the current delegations to exceed the threshold
+// set by validator bond factor
+func (k Keeper) SafelyDecreaseValidatorBond(ctx sdk.Context, validator types.Validator, shares sdk.Dec) error {
+	// If the cap is disabled, do nothing
+	if !k.ValidatorBondFactorEnabled(ctx) {
```

Comment:
I think it would be helpful to count the validator bond because this could be enabled after some periods of disable time.

Reply:
resolved in #78